### PR TITLE
Legal tests react

### PR DESCRIPTION
**Pull request by ashrivastava-qa**
## Because

- For react conversion M1 - simple routes, legal functional tests were needed

## This pull request

- Contains functional tests for legal, terms of service and privacy notice
- Due to an ongoing [bug](https://mozilla-hub.atlassian.net/browse/FXA-6874) for the back button on the Terms of service page not working , the step to test the back button on the test has been commented out.

## Issue that this pull request solves

Closes: #[FXA-6820](https://mozilla-hub.atlassian.net/browse/FXA-6820)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Any other information that is important to this pull request.

[FXA-6820]: https://mozilla-hub.atlassian.net/browse/FXA-6820?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/functional-tests/tests/react-conversion/legal.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+function getReactFeatureFlagUrl(target: BaseTarget, path: string) {
+  return `${target.contentServerUrl}${path}?showReactApp=true`;
+}
+
+test.describe('legal', () => {
+  test('start at legal page', async ({ page, target }) => {
+    await page.goto(getReactFeatureFlagUrl(target, '/legal'));
+
+    // Verify react page has been loaded
+    expect(await page.locator('#root').isVisible()).toBeTruthy();
```
Comment (pdehaan) [inline packages/functional-tests/tests/react-conversion/legal.spec.ts:17]: 👍

File: packages/functional-tests/tests/react-conversion/legal.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  });
+
+  test('start at privacy page', async ({ page, target }) => {
+    await page.goto(getReactFeatureFlagUrl(target, '/legal/privacy'));
+
+    // Verify react page has been loaded
+    expect(await page.locator('#root').isVisible()).toBeTruthy();
+
+    // Verify privacy page is visible
+    await page.waitForTimeout(1000);
```
Comment (pdehaan) [inline packages/functional-tests/tests/react-conversion/legal.spec.ts:69]: Q: Is the pause just for letting the page render?

File: packages/functional-tests/tests/syncV3/syncV3EmailFirst.spec.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ test.describe('Firefox Desktop Sync v3 email first', () => {
 
     //Refresh the page
     await page.reload({ waitUntil: 'load' });
+    await page.waitForTimeout(1000);
```
Comment (pdehaan) [inline packages/functional-tests/tests/syncV3/syncV3EmailFirst.spec.ts:34]: What's this for? Seems unrelated to legal. But wasnt sure if it's fixing some intermittent timing issues or debugging.

**Review by pdehaan — APPROVED** [review]
Do we need/want to add tests for localized versions as well?
Ie: The German version at https://accounts.stage.mozaws.net/de/legal/privacy

If so, I imagine it's probably just copy/paste of "start at privacy page" test and adding `/de/` prefix to URL path and checking for "Firefox Datenschutzhinweis".
Acceptable answers also include "no", "later, file a bug".

**Comment by ashrivastava-qa** [comment]
For localized version testing, my answer is maybe later with a filed ticket or something so that we have a record.

File: packages/functional-tests/tests/react-conversion/legal.spec.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    expect(
+      await page
+        .locator('.privacy-header-policy:has-text("Privacy Notice")')
```
Comment (LZoog) [inline packages/functional-tests/tests/react-conversion/legal.spec.ts:72]: This should instead be the same as the first test, `.font-header:has-text("Firefox Privacy Notice")`. "Privacy Notice" is our default header and doesn't show that the text has been rendered, it will load even if there is an error.
